src/User.php

```diff
@@ -4943,7 +4991,7 @@ public function showItems($tech)
      *
      * @return integer ID of user, 0 if not found nor imported
      */
-    public static function getOrImportByEmail($email = '')
+    public static function getOrImportByEmail($email = '', $createuserfromemail = false)
```

Comment:
I am still not confident with conflicts between `$CFG_GLPI['is_users_auto_add']` and `create_user_from_email` collector configuration.

On previous implementation, a user that exists in LDAP would have been created without using LDAP informations, and I guess it would have lead to unexpected sync issues.

On current implementation, `create_user_from_email` collector configuration will not result in user creation if `$CFG_GLPI["is_users_auto_add"]` is set to false.

So, IMHO, if `create_user_from_email` is set to true, we should use LDAP information if they are available, regardless of `$CFG_GLPI["is_users_auto_add"]`  value, but I would like a confirmation from either @orthagh or @flegastelois . This could be done by using LDAP import using `if ($CFG_GLPI["is_users_auto_add"] || $createuserfromemail)` condition.

Reply:
We discuss about it with @orthagh , and we choose to use your first proposal.
It may generate duplicated users, so we decided to add a tooltip to informa administrators about this.

Changes done on 375398a561fe7b82d02b28fa58c6f262d254e4a6 .